tests/API/Management/ClientsTest.php

```diff
@@ -4,40 +4,141 @@
 use Auth0\SDK\API\Management;
 use Auth0\Tests\API\BasicCrudTest;
 
-class ClientsTest extends BasicCrudTest {
-    
-    protected function getId($entity) {
-        return $entity['client_id'];
-    }
-
-    protected function getApiClient() {
-        $env = $this->getEnv();
-        $token = $this->getToken($env, [
-            'clients' => [
-                'actions' => ['create', 'read', 'delete', 'update']
-            ]
-        ]);
+/**
+ * Class ClientsTest
+ *
+ * @package Auth0\Tests\API\Management
+ */
+class ClientsTest extends BasicCrudTest
+{
+    /**
+     * Unique identifier name for Clients.
+     *
+     * @var string
+     */
+    protected $id_name = 'client_id';
 
-        $this->domain = $env['DOMAIN'];
+    /**
+     * Random number used for unique testing names.
+     *
+     * @var integer
+     */
+    protected $rand;
 
-        $api = new Management($token, $env['DOMAIN']);
+    /**
+     * ClientsTest constructor.
+     */
+    public function __construct()
+    {
+        parent::__construct();
+        $this->rand = rand();
+    }
 
+    /**
+     * Return the Clients API to test.
+     *
+     * @return Management\Clients
+     */
+    protected function getApiClient()
+    {
+        $token = $this->getToken($this->env, [ 'clients' => [ 'actions' => ['create', 'read', 'delete', 'update' ] ] ]);
+        $api = new Management($token, $this->domain);
         return $api->clients;
     }
 
-    protected function getCreateBody() {
-        $client_name = 'test-create-client' . rand();
-        echo "\n-- Using client name $client_name \n";
+    /**
+     * Get the Client create data to send with the test create call.
+     *
+     * @return array
+     */
+    protected function getCreateBody()
+    {
+        return [
+            'name' => 'TEST-CREATE-CLIENT-' . $this->rand,
+            'app_type' => 'regular_web',
+            'sso' => false,
+            'description' => '__Auth0_PHP_initial_app_description__',
+        ];
+    }
+
+    /**
+     * Tests the \Auth0\SDK\API\Management\Clients::getAll() method.
+     *
+     * @param array $created_entity - Entity created during create() test.
+     *
+     * @return mixed
+     *
+     * @throws \Exception
+     */
+    protected function getAllEntities($created_entity)
+    {
+        $fields = array_keys($this->getCreateBody());
+        $fields[] = $this->id_name;
```

Comment:
what's this line doing?

Reply:
Makes sure we have the ID as well so the `findCreatedItem` loop will work properly.